READMEs/erc20-enterprise.md

```diff
@@ -0,0 +1,258 @@
+- [x] Setup : Set envvars
+
+
+## 2. Alice creates a NFT
+In your project folder (i.e. my_project from `Install the library` step) and in the work console where you set envvars, run the following:
+
+Please refer to [datatokens-flow](datatokens-flow.md) and complete the following steps :
+- [x] 2.1 Create an ERC721 data NFT
+
+## 3. Dispenser Flow
+
+In the Python console:
+```python
+# Prepare data for ERC20 Enterprise token
+cap = ocean.to_wei(200)
+from ocean_lib.models.models_structures import CreateErc20Data, DispenserData, ProviderFees, OrderParams
+from ocean_lib.web3_internal.constants import ZERO_ADDRESS
+erc20_data = CreateErc20Data(
+    template_index=2,  # this is the value for ERC20 Enterprise token
+    strings=[
+        "ERC20DT1",
+        "ERC20DT1Symbol",
+    ],  # name & symbol for ERC20 Enterprise token
+    addresses=[
+        alice_wallet.address,  # minter address
+        alice_wallet.address,  # fee manager for this ERC20 Enterprise token
+        alice_wallet.address,  # publishing Market Address
+        ZERO_ADDRESS,  # publishing Market Fee Token
+    ],
+    uints=[cap, 0],
+    bytess=[b""],
+)
+erc20_enterprise_token = nft_token.create_datatoken(
+    erc20_data=erc20_data, from_wallet=alice_wallet
+)
+print(f"ERC20 Enterprise address: {erc20_enterprise_token.address}")
+
+bob_private_key = os.getenv("TEST_PRIVATE_KEY2")
+bob_wallet = Wallet(
+    ocean.web3,
+    bob_private_key,
+    config.block_confirmations,
+    config.transaction_timeout,
+)
+
+# Verify that Bob has ganache ETH
+assert ocean.web3.eth.get_balance(bob_wallet.address) > 0, "need ganache ETH"
+
+# Create & activate dispenser
+dispenser = ocean.dispenser
+dispenser_data = DispenserData(
+    dispenser_address=dispenser.address,
+    allowed_swapper=ZERO_ADDRESS,
+    max_balance=ocean.to_wei(50),
+    with_mint=True,
+    max_tokens=ocean.to_wei(50),
+)
+tx = erc20_enterprise_token.create_dispenser(
+    dispenser_data=dispenser_data, from_wallet=alice_wallet
+)
+assert tx, "Dispenser not created!"
+
+OCEAN_token = ocean.get_datatoken(ocean.OCEAN_address)
+consume_fee_amount = ocean.to_wei(2)
+erc20_enterprise_token.set_publishing_market_fee(
+    publish_market_fee_address=bob_wallet.address,
+    publish_market_fee_token=OCEAN_token.address,  # can be also USDC, DAI
+    publish_market_fee_amount=consume_fee_amount,
+    from_wallet=alice_wallet,
+)
+
+# Approve tokens
+OCEAN_token.approve(
+    spender=erc20_enterprise_token.address,
+    amount=consume_fee_amount,
+    from_wallet=alice_wallet,
+)
+
+# Prepare data for order
+import json
+provider_fees = ocean.build_compute_provider_fees(
+    provider_data=json.dumps({"timeout": 0}, separators=(",", ":")),
+    provider_fee_address=alice_wallet.address,
+    provider_fee_token=OCEAN_token.address,
+    provider_fee_amount=0,
+    valid_until=1958133628,  # 2032
+)
+
+# TODO: this will be handled in web3 py
```

Comment:
no, not here, we just do that in the model. If it will get to the model, it will be converted there.

Reply:
OrderParams already supports ProviderFees as a reusable tuple.